### PR TITLE
Support multiple domain names

### PR DESCRIPTION
**Pull request by RandoMan70**

### CODE REVIEW

**Review by DmitriiPyshinskii — APPROVED** [review]

File: internal/settings/static.go [hunk 1]
```diff
@@ -319,8 +319,8 @@ func (s *Config) validate() error {
 			return xerror.EInternalError("ssl.listen_addr is required", nil)
 		}
 
-		if s.Domain == nil || len(s.Domain.Name) == 0 {
-			return xerror.EInternalError("SSL server is enabled, but no domain name is set", nil)
+		if s.Domain == nil || len(s.Domain.PrimaryName) == 0 {
```
Comment (artemirk) [inline internal/settings/static.go:322]: better 
`s.Domain.PrimaryName == ""`

File: pkg/xhttp/domain_config.go [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+import (
+	adminAPI "github.com/vpnhouse/api/go/server/tunnel_admin"
+	"github.com/vpnhouse/tunnel/pkg/xerror"
+)
+
+// DomainConfig is the YAML version of the `adminAPI.DomainConfig` struct.
+type DomainConfig struct {
+	Mode        string   `yaml:"mode" valid:"required"`
+	PrimaryName string   `yaml:"name" valid:"dns,required"`
```
Comment (artemirk) [inline pkg/xhttp/domain_config.go:11]: Why it's named as `PrimaryName` Can it be simple `Name` as yaml defines it's the `name`

File: pkg/xhttp/domain_config.go [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+	}
+	if len(c.Mode) == 0 {
+		return xerror.EInternalError("domain.mode is required", nil)
+	}
+	if _, ok := modes[c.Mode]; !ok {
+		return xerror.EInternalError("domain.mode got unknown value, expecting `direct` or `reverse-proxy`", nil)
+	}
+
+	if c.Mode == string(adminAPI.DomainConfigModeReverseProxy) {
+		if len(c.Schema) == 0 {
```
[the remark this answers is not in the document]
Comment (artemirk) [reply pkg/xhttp/domain_config.go:42]: Worth to have check with  c.Schema == "http" || c.Schema == "https"  ?

File: pkg/xhttp/domain_config.go (hunk 2 above)
Comment (RandoMan70) [reply pkg/xhttp/domain_config.go:11]: PrimaryName is better for reading code. "name" in yaml is left for backward compatibility.

File: pkg/xhttp/domain_config.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (RandoMan70) [reply pkg/xhttp/domain_config.go:42]: I didn't change this code, just moved to appropriate place. Let leave as is for now.

**Pull request merged**